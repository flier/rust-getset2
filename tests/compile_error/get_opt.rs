use getset2::Getter;

#[derive(Getter)]
pub struct Struct {
    #[get(opt)]
    field: usize,
}

fn main() {}
