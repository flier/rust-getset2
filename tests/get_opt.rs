use std::collections::HashMap;

use getset2::Getter;

#[derive(Default, Getter)]
struct Foo {
    #[get(opt, mut)]
    pub option_field: Option<HashMap<String, usize>>,
}

#[test]
fn get_opt() {
    let mut foo = Foo::default();

    let _ = foo.option_field.insert(HashMap::new());

    foo.option_field_mut()
        .unwrap()
        .insert("foo".to_owned(), 123);

    assert_eq!(foo.option_field().unwrap().get("foo"), Some(&123));
}
