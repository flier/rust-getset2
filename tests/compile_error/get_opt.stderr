error: field with #[get(opt)] should be an Option type
 --> tests/compile_error/get_opt.rs:6:12
  |
6 |     field: usize,
  |            ^^^^^
